plugins/toml/core/src/main/kotlin/org/toml/lang/psi/impl/Psi.kt

```diff
@@ -36,7 +38,10 @@ class TomlKeyValueImpl(type: IElementType) : CompositePsiElement(type), TomlKeyV
 
 class TomlKeySegmentImpl(type: IElementType) : CompositePsiElement(type), TomlKeySegment {
 
-    override fun getName(): String = text
+    override fun getName(): String = when (val child = node.findChildByType(TOML_SINGLE_LINE_STRINGS)) {
+        null -> text
+        else -> offsetsForTomlText(child).value?.substring(child.text)?.unescapeToml(child.elementType) ?: text
```

Comment:
Actually, it's very similar to `org.toml.lang.psi.ext.TomlLiteralKind.String#getValue`
Maybe reuse it not to duplicate code?
Something like
```suggestion
      else -> {
          val kind = TomlLiteralKind.fromAstNode(child) as? TomlLiteralKind.String ?: return text
          kind.value ?: text
      }
```

Reply:
Good idea, thank you.